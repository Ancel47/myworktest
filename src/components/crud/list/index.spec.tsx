import React from "react";

import {
    render,
    TestWrapper,
    MockJSONServer,
    waitForElementToBeRemoved,
} from "@test";

import { Table, Column } from "@components";

import { List } from "./index";

describe("<List/>", () => {
    describe("JSON Rest Server", () => {
        it("mounts with table", async () => {
            const { getByText } = render(
                <List key="posts" resourceName="posts">
                    <Table rowKey="id" />
                </List>,
                {
                    wrapper: TestWrapper({
                        dataProvider: MockJSONServer,
                        resources: [{ name: "posts" }],
                    }),
                },
            );

            await waitForElementToBeRemoved(() => getByText("No Data"));
        });
        it("renders given data", async () => {
            const { container } = render(
                <List key="posts" resourceName="posts">
                    <Table rowKey="id">
                        <Column key="title" title="Title" dataIndex="title" />
                    </Table>
                </List>,
                {
                    wrapper: TestWrapper({
                        dataProvider: MockJSONServer,
                        resources: [{ name: "posts" }],
                    }),
                },
            );

            expect(container).toMatchSnapshot();
        });

        it("should wrap with given component", async () => {
            const { container } = render(
                <List component={"section"} resourceName="posts">
                    <Table rowKey="id">
                        <Column key="title" title="Title" dataIndex="title" />
                    </Table>
                </List>,
                {
                    wrapper: TestWrapper({
                        dataProvider: MockJSONServer,
                        resources: [{ name: "posts" }],
                    }),
                },
            );
            expect(container.querySelector("section")).toBeTruthy();
        });
    });
});
