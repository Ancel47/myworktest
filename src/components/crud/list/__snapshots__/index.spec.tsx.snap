// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> JSON Rest Server renders given data 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Posts
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
      style="padding: 0px;"
    >
      <div
        class="ant-table-wrapper"
        style="width: 100%;"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div>
            <div
              class="ant-spin ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Title
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          Necessitatibus necessitatibus id et cupiditate provident est qui amet.
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          Recusandae consectetur aut atque est.
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-center"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
